### PR TITLE
KWS-3079, KWS-3080, KWS-3081, KWS-3082 Alerts

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
https://jira.superawesome.tv/browse/KWS-3079
https://jira.superawesome.tv/browse/KWS-3080
https://jira.superawesome.tv/browse/KWS-3081
https://jira.superawesome.tv/browse/KWS-3082

## Description
* Built AlertsView
* Built AlertsDataProvider
* Built AlertsViewModel
* Built AlertViewModel
* Built AlertCells
* Connected PJ Data SDK for Alerts
* Added Tests for AlertsViewModel
* Added Tests for AlertViewModel
* Added Tests for AlertsDataProvider
* Added Shake Placeholder Screen
* Connected Alert Actions for tapping UserAvatar and buttons

## Screenshots
simulator screen shot - iphone 7 - 2018-06-14 at 13 28 05

simulator screen shot - iphone 7 - 2018-06-14 at 13 28 10

### CODE REVIEW

File: Example/Tests/AlertsView/DataProvider/AlertsDataProviderTests.swift [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    // MARK: Constants
+    
+    private let fakeLoggedUser = LoggedUserModel(token: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJ1c2VySWQiOjExMTU5MDQsImFwcElkIjoyLCJjbGllbnRJZCI6InBvcGphbSIsInNjb3BlIjoidXNlciIsImlhdCI6MTUyMjc1NTA5NSwiZXhwIjoxODM4MTE1MDk1LCJpc3MiOiJzdXBlcmF3ZXNvbWUifQ.NdqQ03LE1VHIV_LCmUb3oB1TQHwpwiDiEk2dmFUC5hHFTj8pEtRq5BRSF6fpZVRJqFNmPGzZ1bzrRKF8sUSP_c_vfA_knBUhAT2XmWAznpjhqQKnjg2wGWr0NqlyiVNCY-h0U9VbdSeWZ3Jx401xmOylIejmYXGGq7jeeuuwyq798iPQ55HP_skFDh_0_Z8fKfzpKa0gMiKHteIroI8eap9as-IRDp6aDRi2tk2tNHOrzbmjiNnMZ8N7xs36X1Lw3-JWgpC8OkKRXeOzFCxPm_MjCo9RoT02Qh-Z3clVu6pbmNMtJPeHj9yEpGFubG1t6q0fCgkir9UhUMZUU74b9Q", id: "7d0e17cb-7c1f-4577-9504-a0bf4fcd7a41")
+    
+    // MARK: Properties
+    
+    var dataProvider: AlertsDataProvider!
+    var failProvider: AlertsDataProvider!
+    var failFollowProvider: AlertsDataProvider!
+    // MARK: Methods
```
Comment (Tbruni85) [inline Example/Tests/AlertsView/DataProvider/AlertsDataProviderTests.swift]: space please..

File: PopJamUI/Assets/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -87,6 +87,25 @@
...
+"AlertsSkakeKey" = "Shake at 6pm tonight";
+"AlertsTopFriendKey" = "added you as a Top Friend";
+"AlertsCommentedOnPostKey" = "commented on your post";
+"AlertsFollowButtonTitleKey" = "FOLLOW";
+"AlertsFollowingButtonTitleKey" = "FOLLOWING";
+"AlertsViewButtonTitleKey" = "VIEW";
+"AlertsShakeButtonTitleKey" = "SHAKE";
+
+"AlertsLevelAchievedTitleKey" = "LEVEL %i ACHEIVED!";
+"AlertsLevelAchievedMessageKey" = "You unlocked:";
```
Comment (Tbruni85) [inline PopJamUI/Assets/en.lproj/Localizable.strings]: Move the emojis in the localised files

File: PopJamUI/Classes/Alerts/ViewModel/AlertViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    private func attributedDescription(withFont font: UIFont) -> NSAttributedString {
+        guard let name = model?.name else { return NSAttributedString(string: "", attributes: [.font: font]) }
+        var alertText = ""
+        var attachmentString: NSAttributedString?
+        var textColor = Color.textColour1.value
+        switch name {
+        case .userLikedYourPost:
+            alertText = "❤️ \("AlertsLikePostKey".defaultLocalisation())"
```
Comment (Tbruni85) [inline PopJamUI/Classes/Alerts/ViewModel/AlertViewModel.swift]: move these to the localised file

File: Example/PopJamUI/Protocols/CanShowUserProfile.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+//  CanShowUserProfile.swift
+//  PopJamUI_Example
+//
+//  Created by Myles Eynon on 13/06/2018.
+//  Copyright © 2018 CocoaPods. All rights reserved.
+//
+
+import PopJamUI
+
+public protocol CanShowUserProfile: ViewAvatarDelegate {
```
Comment (GuilhermeMota93) [inline Example/PopJamUI/Protocols/CanShowUserProfile.swift]: Being this a `protocol` shouldn't the name of the protocol reference the type? Something in the lines of `CanShowUserProfileProtocol` ?

File: Example/PopJamUI/ServiceProvider.swift [hunk 5]
```diff
@@ -32,7 +32,7 @@ public class ServiceProvider {
     private var eventsProvider: SocialSDK?
     private let eventsManager: EventManager = EventManager.shared
 
-    public var environment: Environment = .staging {
+    public var environment: Environment = .production {
```
Comment (GuilhermeMota93) [inline Example/PopJamUI/ServiceProvider.swift]: Just double checking if `production` is what we want as default here.

File: PopJamUI/Assets/de.lproj/Localizable.strings [hunk 6]
```diff
@@ -87,6 +87,24 @@
...
+"AlertsCommentedOnPostKey" = "💬 commented on your post";
+"AlertsFollowedKey" = "followed you";
+"AlertsLikeCommentKey" = "❤️ your comment";
+"AlertsLikePostKey" = "❤️ your post";
+"AlertsLevelAcheivedKey" = "🎉 LEVEL %i ACHIEVED! 🎉";
+"AlertsMentionCommentKey" = "📣 mentioned you in a comment";
+"AlertsMentionPostKey" = "📣 mentioned you in a post";
+"AlertsAlmostAtLevelKey" = "⭐️ You are 1 follower from Level %i ⭐️";
+"AlertsRejammedPostKey" = "rejammed your post";
+"AlertsSkakeKey" = "👋 Shake at 6pm tonight";
```
Comment (GuilhermeMota93) [inline PopJamUI/Assets/de.lproj/Localizable.strings]: I'm completely out of context here, and please ignore if absurd, but should `6pm` be hardcoded? Just wondering if this time changes in the future. I would imagine to be a possible formatted string as the `AlertsAlmostAtLevelKey` for the levels.

File: PopJamUI/Classes/Alerts/DataSource/AlertsDataSource.swift [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        delegate?.updateCollection()
+    }
+    
+    public func didFetchData(atIndex index: Int, data: Any, viewModel: ViewModel) {
+        let indexPath = IndexPath(row: index, section: mainSection)
+        delegate?.updateCell(atIndexPath: indexPath)
+    }
+    
+    public func didFetchDataWithError(error: Error, viewModel: ViewModel) {
+        print("Error fetching alerts \(error)")
```
Comment (GuilhermeMota93) [inline PopJamUI/Classes/Alerts/DataSource/AlertsDataSource.swift]: Are we keeping prints?

File: Example/PopJamUI/ServiceProvider.swift (hunk 5 above)
Comment (me2k) [reply Example/PopJamUI/ServiceProvider.swift]: good spot!

File: PopJamUI/Classes/Alerts/DataSource/AlertsDataSource.swift (hunk 7 above)
Comment (me2k) [reply PopJamUI/Classes/Alerts/DataSource/AlertsDataSource.swift]: here, yes

File: PopJamUI/Assets/de.lproj/Localizable.strings (hunk 6 above)
Comment (devgabrielcoman) [reply PopJamUI/Assets/de.lproj/Localizable.strings]: Defo not hardcoded. It's provided in the Alert information coming from the Data SDK as:
hour: Int
minute: Int
and you can build the actual time from there.

**Pull request merged**